lib/ecto/adapters/sql.ex

```diff
@@ -1034,4 +1042,13 @@ defmodule Ecto.Adapters.SQL do
   defp sql_color("begin" <> _), do: :magenta
   defp sql_color("commit" <> _), do: :magenta
   defp sql_color(_), do: nil
+
+  defp parse_param({binary, Postgrex.Extensions.UUID}) do
+    case Ecto.UUID.load(binary) do
```

Comment:
I'd like to point out that TDS adapter uses it own [custom](https://github.com/elixir-ecto/ecto_sql/blob/c421741174f97d871da5ece9955629261282bde4/lib/ecto/adapters/tds.ex#L61-L65) implementation for UUIDs

Reply:
Won't this implementation only try to parse UUID values originating from PG database? It matches on `Postgrex.Extensions.UUID` type.

Reply:
We shouldn’t hardcode it to Postgrex though, as the adapters are meant to be general. Not sure how to tackle it though...

Reply:
Right loading TDS UUID using `Ecto.UUID` is incompatible:

``` 
a = Tds.Ecto.UUID.bingenerate()
Tds.Ecto.UUID.load(a) == Ecto.UUID.load(a) # => false
```

Since we don't want to reference `Postgrex` inside the adapter, maybe defining an optional method on connection object could do the trick? I've updated the PR.